baya/membership.py

```diff
@@ -35,33 +35,44 @@ def __rxor__(self, other):
 
     # Comparisons.
     def __eq__(self, other):
-        return self.__class__ is other.__class__
+        raise NotImplementedError()
```

Comment:
IMO, the old implementation was not something you'd ever want from equality semantics, though it was overridden by child classes in each case.